### ISSUE TITLE
payment_event vs order.set_status

### ISSUE THREAD

**Issue by aleprovencio** [daneoshiga/django-oscar-pagseguro#3]
Tudo jóia @daneoshiga?

Queria saber se existe alguma razão de os sinais da oscar-pagseguro setarem o
status do pedido ao invés de emitirem eventos de pagamento com os estados do
pagseguro ('aguardando pagemento', 'pago', etc).

Pensei nisto por que apenas o pagseguro pode mudar estes estados no fluxo
normal, se fosse alterado diretamente pelo dashboard da aplicação o pagseguro
não reconheceria, não é uma comunicação de 2 vias certo?

Se o oscar-pagseguro fosse implementado paga emitir sinais de pagamento, o
estado do pedido poderia ser alterado automaticamente via EventHandlers, mas
também manualmente pelo dashboard da aplicação no caso de algum problema com o
pagseguro/correios/etc.

O que acha?

**Comment by daneoshiga** [comment]
Eventos?
Na verdade, se não me engano, é possível mudar esses status diretamente na interface administrativa do Oscar sim, agora não entendi o que você quis dizer com eventos, ler os sinais e emitir outro sinal?

**Comment by aleprovencio** [comment]
Sim, eventos! Ao ler o sinal do pagseguro, ao invés de "order.set_status('Aguardando pagamento')", usar um add_payment_event('Aguardando pagamento').  No dashboard é possível também criar um evento de pagamento, mas o mais intuitivo dele alterar é o estado do pedido.

Veja bem, só estou tentando entender a pipeline, na forma atual:

```
- cliente faz a compra
- o pagseguro aprova e gera um sinal pra aplicação
- a aplicação seta o estado do pedido para "Pago"
- o admin do dashboard procura os pedidos do dia que foram pagos e prepara o frete
- o admin despacha o produto e emite um evento de frete via dashboard
- o admin admite pedidos com estado "Disponível" e "Cancelado" como pedidos finalizados e fica atento a todos os pedidos com outros estados
```

Aparentemente não há problemas, mas olhando os eventos de pagamento, o único que existe até então é um "pre-auth" de quando o pagseguro está aprovando a transação. Será que não seria melhor estes sinais do pagseguro serem eventos de pagamento e o admin do dashboard ficar de olho no estado do pedido e alterá-los quando necessário?

**Comment by daneoshiga** [comment]
Ah entendi

Eu realmente não me atentei ao funcionamento dos eventos do Oscar, o pouco que olhei não resolveu meu problema, acabei abandonando.

Quando tiver um tempo dou uma olhada, ou se quiser, faça um pull-request :+1:

**Comment by aleprovencio** [comment]
OK, qualquer novidades postamos aqui
